invenio_rdm_records/services/schemas/files.py

```diff
@@ -1,53 +1,64 @@
 # -*- coding: utf-8 -*-
 #
-# Copyright (C) 2020-2021 CERN.
+# Copyright (C) 2020-2023 CERN.
 # Copyright (C) 2020-2021 Northwestern University.
 #
 # Invenio-RDM-Records is free software; you can redistribute it and/or modify
 # it under the terms of the MIT License; see LICENSE file for more details.
 
 """RDM record schemas."""
 
-from marshmallow import Schema, fields
-from marshmallow_utils.fields import SanitizedUnicode
+from marshmallow import Schema, fields, pre_load
+from marshmallow_utils.fields import NestedAttribute, SanitizedUnicode
 from marshmallow_utils.permissions import FieldPermissionsMixin
 
 
 class FileSchema(Schema):
     """File schema."""
 
-    type = fields.String()
     checksum = fields.String()
-    size = fields.Integer()
+    ext = fields.String(attribute="file.ext")
     key = SanitizedUnicode()
-    version_id = SanitizedUnicode()
-    bucket_id = SanitizedUnicode()
-    mimetype = SanitizedUnicode()
-    storage_class = SanitizedUnicode()
-
-    # TODO (Alex): See how this fits with using the refactored Linker
-    # links = fields.Method('get_links')
-
-    # def get_links(self, obj):
-    #     """Get links."""
-    #     return {
-    #         'self': api_link_for(
-    #             'object', bucket=obj['bucket'], key=obj['key'])
-    #     }
+    metadata = fields.Dict()
+    mimetype = fields.String(attribute="file.mimetype")
+    size = fields.Integer(attribute="file.size")
 
 
 class FilesSchema(Schema, FieldPermissionsMixin):
     """Files metadata schema."""
 
     field_dump_permissions = {
+        "count": "read_files",
         "default_preview": "read_files",
+        "entries": "read_files",
         "order": "read_files",
+        "total_bytes": "read_files",
     }
 
     enabled = fields.Bool()
     default_preview = SanitizedUnicode(allow_none=True)
     order = fields.List(SanitizedUnicode())
 
+    count = fields.Integer(dump_only=True)
+    total_bytes = fields.Integer(dump_only=True)
+
+    entries = fields.Dict(
+        keys=SanitizedUnicode(),
+        values=NestedAttribute(FileSchema),
+        dump_only=True,
+    )
+
+    @pre_load
+    def clean(self, data, **kwargs):
```

Comment:
I've seen this "utility" method somewhere (IIRC related to custom fields). It might be worth making it into a Mixin. Maybe in Marshmallow-Utils?

Reply:
I think it's in RecordBase I think in Invenio-Records-Resources.